main.tf

```diff
@@ -1,116 +1,116 @@
 data "aws_ami" "ami" {
   most_recent = true
-  owners = ["099720109477"]
+  owners      = ["099720109477"]
   filter {
-    name = "name"
-    values = ["ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-*"]
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*"]
   }
   filter {
-    name = "virtualization-type"
+    name   = "virtualization-type"
     values = ["hvm"]
   }
 }
 
 resource "aws_instance" "instance" {
-  ami = "${coalesce(var.ami, data.aws_ami.ami.image_id)}"
-  instance_type = "${var.instance_type}"
-  key_name = "${var.ssh_key_name}"
-  security_groups = ["${aws_security_group.sg.name}"]
+  ami                         = coalesce(var.ami, data.aws_ami.ami.image_id)
+  instance_type               = var.instance_type
+  key_name                    = var.ssh_key_name
+  security_groups             = [aws_security_group.sg.name]
   associate_public_ip_address = true
+  iam_instance_profile        = aws_iam_instance_profile.profile.name
 
-  tags {
-    Name = "${var.project_name}"
+  tags = {
+    Name = var.project_name
   }
 
   root_block_device {
-    volume_size = "${var.volume_size}"
+    volume_size = var.volume_size
   }
 
   connection {
-    user = "ubuntu"
-    private_key = "${file("${var.ssh_private_key}")}"
+    host        = self.public_ip
+    user        = "ubuntu"
+    private_key = file(var.ssh_private_key)
   }
 
   provisioner "file" {
-    content = "${file("docker-compose.yml")}"
-    destination = "~/docker-compose.yml"
-  }
-
-  provisioner "file" {
-    content = "${var.init_script}"
-    destination = "~/init.sh"
+    source      = coalesce(var.cloudwatch_agent_config, "${path.module}/cwagentconfig.json")
+    destination = "~/cwagentconfig"
   }
 
   provisioner "remote-exec" {
     inline = [
-      "mkdir -p ~/config"
+      "sudo apt-get update",
+      "sudo DEBIAN_FRONTEND=noninteractive apt-get upgrade -y -o DPkg::options::=\"--force-confdef\" -o DPkg::options::=\"--force-confold\"", "sudo apt-get install -y docker.io docker-compose",
+      "sudo usermod -aG docker $USER"
     ]
   }
 
-  provisioner "file" {
-    source = "${path.cwd}/config/"
-    destination = "~/config/"
-  }
-
   provisioner "remote-exec" {
     inline = [
```

Comment:
Nit: would it be OK to merge this provisioner into the previous one?

Reply:
Of course